src/Pkg.jl

```diff
@@ -61,7 +61,7 @@ const UPDATED_REGISTRY_THIS_SESSION = Ref(false)
 const OFFLINE_MODE = Ref(false)
 const RESPECT_SYSIMAGE_VERSIONS = Ref(true)
 # For globally overriding in e.g. tests
-const DEFAULT_IO = Ref{Union{IO, Nothing}}(nothing)
+const DEFAULT_IO = Base.ScopedValues.ScopedValue{IO}()
```

Comment:
@Keno I suspect that this line broke PkgEval. See https://github.com/JuliaLang/julia/commit/1067db80b96cafb9da6864afbfdd9d6e27665c33 for further discussion.

Essentially, I think it broke this section in the PkgEval script: https://github.com/JuliaCI/PkgEval.jl/blob/2b001d04dde0c33a22680ce76d22518a2b31de3e/scripts/evaluate.jl#L54-L55

```julia
    io = IOBuffer()
    Pkg.DEFAULT_IO[] = io
```

What's the best way to fix this? Modify PkgEval to check the type of `Pkg.DEFAULT_IO` and branch accordingly?